packages/api/endpoints/rules.js

```diff
@@ -108,31 +108,36 @@ async function post(req, res) {
  */
 async function put({ params: { name }, body }, res) {
   const model = new models.Rule();
+  const apiRule = { ...body };
   let newRule;
 
-  if (name !== body.name) {
+  if (name !== apiRule.name) {
     return res.boom.badRequest(`Expected rule name to be '${name}', but found`
       + ` '${body.name}' in payload`);
   }
 
   try {
     const oldRule = await model.get({ name });
     const dbClient = await getKnexClient();
-
+    apiRule.updatedAt = Date.now();
+    apiRule.createdAt = oldRule.createdAt;
     // If rule type is onetime no change is allowed unless it is a rerun
-    if (body.action === 'rerun') {
+
+    if (apiRule.action === 'rerun') {
       return models.Rule.invoke(oldRule).then(() => res.send(oldRule));
     }
 
-    const fieldsToDelete = Object.keys(oldRule).filter((key) => !(key in body));
-    const newPostgresRecord = await translateApiRuleToPostgresRule(body, dbClient);
+    const fieldsToDelete = Object.keys(oldRule).filter(
+      (key) => !(key in apiRule) && key !== 'createdAt'
+    );
+    const newPostgresRecord = await translateApiRuleToPostgresRule(apiRule, dbClient);
 
     await dbClient.transaction(async (trx) => {
       await trx(tableNames.rules)
```

Comment:
This code was written prior to the existence of the DB models: but again, I would like to see this moved to an `upsert` method of the rule PG model

Reply:
d2ee549eb